### ISSUE TITLE
Unable to save data (eg Friendly Name) in Hebrew

### ISSUE THREAD

**Issue by mail955** [dOpensource/dsiprouter#315]
Thank you so much for your great system :)

I am trying to save text in Hebrew, for example Friendly Name in the name of a server or in mapping numbers, and the following error occurs:
`Server Issue
Detailed Information: 'charmap' codec can't encode characters in position 5-7: character maps to <undefined>
`
And this is the output of
`tail -f /var/log/dsiprouter.log`

```
[2020-12-06 13:45:04] [22330-> 140209780553536] [INFO]: INSERT INTO dr_rules (groupid, prefix, timerec, routeid, gwlist, description) VALUES (% s,% s,% s,% s,% s,% s)
[2020-12-06 13:45:04] [22330-> 140209780553536] [INFO]: INSERT INTO dr_rules (groupid, prefix, timerec, routeid, gwlist, description) VALUES (% s,% s,% s,% s,% s,% s)
[2020-12-06 13:45:04] [22330-> 140209780553536] [INFO]: (9000, '9723', '', '', '# 18', 'name: טסט')
[2020-12-06 13:45:04] [22330-> 140209780553536] [INFO]: (9000, '9723', '', '', '# 18', 'name: טסט')
[2020-12-06 13:45:04] [22330-> 140209780553536] [INFO]: ROLLBACK
[2020-12-06 13:45:04] [22330-> 140209780553536] [INFO]: ROLLBACK
[2020-12-06 13:45:04] [22330-> 140209780553536] [ERROR]: (((EXCEPTION)))
[CLASS]: <class 'UnicodeEncodeError'>
[VALUE]: 'charmap' codec can not encode characters in position 5-7: character maps to <undefined>
(((BACKTRACE)))
[FILE]: /opt/dsiprouter/gui/dsiprouter.py
[LINE NUM]: 1297
[FUNCTION]: addUpdateInboundMapping ()
[SOURCE]: db.commit ()
```

If I change the value in MySql, then there it is kept correct, but in the interface you see question marks (???) and if you try to change it and thread then even in such a case comes the above error

I would be very happy to help what I am missing.

The system is installed in a clean installation on CentOS installed for this purpose.

Thanks :)
![Screenshot of dSIPRouter Error Page](https://user-images.githubusercontent.com/26262274/101281979-010e1e00-37db-11eb-97e6-3c5822dd110c.jpg)

**Comment by devopsec** [comment]
This might be an sqlalchemy issue, since you can insert directly in the database without issue (UTF8 supports hebrew characters)  
Will look further into this and get an answer to you

**Comment by devopsec** [comment]
Seems like we might need to encode the strings in UTF-8 before passing the query to sqlalchemy.  
For Example:

```
cd /opt/dsiprouter/gui

python3 <<'EOF'
from database import db_engine, SessionLoader, GatewayGroups
db = SessionLoader()
try:
    db.query(GatewayGroups).filter(GatewayGroups.id == 2).update({'description':'name: טסט'})
    db.commit()
except Exception as ex:
    print('ERROR: {}'.format(str(ex)))
    db.rollback()
    db.flush()
finally:
    db.close()
EOF
```

This will give a unicode error wheras:

```
cd /opt/dsiprouter/gui

python3 <<'EOF'
from database import db_engine, SessionLoader, GatewayGroups
db = SessionLoader()
try:
    db.query(GatewayGroups).filter(GatewayGroups.id == 2).update({'description':'name: טסט'.encode('utf-8')})
    db.commit()
except Exception as ex:
    print('ERROR: {}'.format(str(ex)))
    db.rollback()
    db.flush()
finally:
    db.close()
EOF
```

Will update the DB.  
But the return from a select does not seem right:

```
cd /opt/dsiprouter/gui

python3 <<'EOF'
from database import db_engine, SessionLoader, GatewayGroups
db = SessionLoader()
try:
    x = db.query(GatewayGroups).filter(GatewayGroups.id == 2).first().description
    print(x)
except Exception as ex:
    print('ERROR: {}'.format(str(ex)))
finally:
    db.close()
EOF
```

Will output:

```
name: ×˜×¡×˜
```

**Comment by devopsec** [comment]
Doing some more digging it seems that setting the charset to utf8mb4 when the database driver is created is required.

**Comment by devopsec** [comment]
Resolved in 97d27d1c2c41e6e80bbf8d6bb33340e45a38c2d7
